rfcs/2021-11-whcm-support.md

```diff
@@ -0,0 +1,108 @@
+
+Windows High Contrast Mode (WHCM) is a unique accessibility feature originally designed for Windows users with contrast sensitivity issues. Other reasons a user might enable WHCM include:
+
+- Reducing visual noise to improve focus
+- Easing eye strain, migraines, or light sensitivity
+- Preferring a specific color scheme
+
+With WHCM, users can select a theme of colors for a scoped number of semantic elements. This theme is then applied to user interfaces and applications.
+
+Windows provides a handful of preexisting themes, which can be customized to the user's preference. The WHCM color palette consists of the following:
+
+- Text (foreground) color
+- Hyperlink color
+- Disabled text color
+- Selected text color
+  - Text (foreground) color
+  - Background color
+- Button text
+  - Text (foreground) color
+  - Background color
+- Background color
+
+## Problem
+
+Our [Design System](https://design.cms.gov/) was built primarily by developers using macOS, so it wasn't as apparent how our UI library looked or behaved when WHCM was enabled. As a result, many of our more complex UI components range from difficult to impossible to use.
+
+[This initial full audit captures how these components appear/behave in WHCM](https://docs.google.com/document/d/1uzApahaUse04UITNFNa-uFY4IQ3Z9mAuzx5xNmniUJ0/edit?usp=sharing).
+
+Outages can be categorized as follows:
+
+- Inconsistent disabled states
+  - Should use the mapped WHCM "disabled text color"
+  - Needs `not-allowed` cursor for additional context
+- Form fields lack usable success, error, and hint text/states
+  - Due to the limited color palette...
+    - Field error and success states look indistinguishable from default field state
+    - Hint and error text look identical to each other
+    - It's not entirely obvious an error has happened in a form field without reading the text
+  - The solution is to avoid relying on color alone when conveying different messages/states (_needs a11y/design direction_)
+    - Different border styles/widths?
+    - Use of italic or bold text?
+    - Label error text with "Error:" language?
+  - Seen in all form fields, but mostly inherited from the `FormLabel` component
+- Icons vary in visibility
+  - Some icons don't render
+  - Icons that do render don't reflect WHCM color palette and are difficult to see
+  - Reliance on icons only to convey meaning is lost, especially for Alerts, Buttons with icons, and displaying active states in navigation
+  - Icons should inherit the appropriate text color as mapped to the WHCM theme
+- Border issues
+  - Thin/no borders make it hard to see Badge, FilterChip, HelpDrawerToggle, Tooltip, and the Modal close button
+  - Content in Accordion and Tabs aren't visually grouped together as they are in non-WHCM
+- Buttons (_needs a11y/design direction_)
+  - Use of "ghost style" buttons results in weird padding/margins and/or broken borders
+  - Should buttons and links styled as buttons look the same?
+
+## Proposal
```

Comment:
A few questions:
1. Apart from the technical implementations for WHCM, is design/UX/accessibility expert input needed to implement fixes?
2. It is unclear to me how much time it would take to implement updates for this. What kind of order would you recommend implementing changes? Would you update for WHCM one component at a time? Or would you address multiple components and focus on resolving/updating each outage category ?